Source/WebCore/page/csp/ContentSecurityPolicy.cpp

```diff
@@ -776,6 +776,41 @@ AllowTrustedTypePolicy ContentSecurityPolicy::allowTrustedTypesPolicy(const Stri
         : details;
 }
 
+bool ContentSecurityPolicy::requireTrustedTypesForSinkGroup(const String& sinkGroup) const
+{
+    bool required = false;
+    for (auto& policy : m_policies) {
+        if (policy->requiresTrustedTypesForScript() && sinkGroup == "script"_s) {
+            required = true;
+            break;
+        }
+    }
+    return required;
+}
+
+bool ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup(const String& stringContext, const String& sink, const String& sinkGroup, const String& source) const
+{
+    String sourceURL;
+    TextPosition sourcePosition(OrdinalNumber::beforeFirst(), OrdinalNumber());
+
+    bool isAllowed = true;
+    for (auto& policy : m_policies) {
+        if (policy->requiresTrustedTypesForScript() && sinkGroup == "script"_s) {
+            if (!policy->isReportOnly())
+                isAllowed = false;
+
+            String consoleMessage = makeString(policy->isReportOnly() ? "[Report Only] " : "",
+                "This requires a "_s, stringContext, " value else it violates the following Content Security Policy directive: \"require-trusted-types-for 'script'\""_s);
+
+            const ContentSecurityPolicyDirectiveList& list = *policy.get();
+
+            String sample = makeString(sink, "|"_s, StringView(source).left(40).toString());
```

Comment:
Why 40?

Reply:
That's how it's specced. Step 2.7 here https://w3c.github.io/trusted-types/dist/spec/#should-block-sink-type-mismatch

Reply:
Can we add a link to the algorithm just before ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup?

Reply:
Also is the toString() call needed?